clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -22,11 +22,11 @@
     private final String topic;
     private final int partition;
     private final long offset;
-    private final long timestamp;
+    private final Long timestamp;
     private final TimestampType timestampType;
-    private final long checksum;
-    private final int serializedKeySize;
-    private final int serializedValueSize;
+    private final Long checksum;
+    private final Integer serializedKeySize;
+    private final Integer serializedValueSize;
```

Comment:
This is also related to the following PR https://github.com/apache/kafka/pull/1015 that uses sentinels in a number of places. It would be nice to be consistent on -1 versus null.